### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by Fourteen98**
### New Features: 

-Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
image

- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
image

- Update all animals' weights that are negative to be their weight multiplied by -1.
image

- How many  animals are there?
image

- How many animals have never tried to escape?
image

- What is the average weight of animals?
image

- Who escapes the most, neutered or not neutered animals?
image

- What is the minimum and maximum weight of each type of animal? 
image

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018-11-15', 2, True, 5.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021-01-07', 1, False, 15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017-05-12', 5, True, 11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', '2021-11-15', 2, True, -5.7);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', '2020-02-08', 0, False, -11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', '1993-04-02', 3, False, -12.13);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', '2005-06-12', 1, True, 45);
```
Comment (cynthiainga) [inline data.sql]: - [x] Please kindly change this `weight_kg`, is also a negative number `-45` 😉 😉

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Boarmon', '2005-06-07', 7, True, 20.4);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Blossom', '1998-10-13', 3, False, 17);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Ditton', '1999-08-31', 2, False, 16.3);
```
Comment (cynthiainga) [inline data.sql]: - [x] The neutered of those two values is `true` and not `false`, please kindly change it 🙃 🙃

File: queries.sql [hunk 3]
```diff
@@ -12,7 +12,46 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 
 SELECT * FROM animals WHERE neutered = true;
 
-SELECT * FROM ANIMALS WHERE name NOT LIKE '%Gabumon%';
+SELECT * FROM animals WHERE name NOT LIKE '%Gabumon%';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+UPDATE animals SET species = 'unspecified' WHERE species IS NULL;
```
Comment (cynthiainga) [inline queries.sql]: - [x] Please kindly always start the transaction with `BEGIN;` 😉
- [x] Also select species from animals after the update to verify if the change was made 🙃

File: queries.sql [hunk 4]
```diff
@@ -12,7 +12,46 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+ROLLBACK;
+
```
Comment (cynthiainga) [inline queries.sql:24]: - [x] Please kindly select species from animals after the rollback to verify if the change was made 🙃

File: queries.sql [hunk 5]
```diff
@@ -12,7 +12,46 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
```
Comment (cynthiainga) [inline queries.sql:31]: - [x]  Please kindly always start the transaction with `BEGIN;` 😉
- [x] Please select species from animals after the update to verify if the change was made 🙃
- [x] Also kindly select species from animals after the commit to verify if the change was made 🙃

File: queries.sql [hunk 6]
```diff
@@ -12,7 +12,46 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT animals;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg > 0 OR weight_kg  > 0;
+
+ROLLBACK TO animals;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
```
Comment (cynthiainga) [inline queries.sql:50]: - [x] Please kindly start the transaction with `BEGIN;` 😉
- [x] Also please commit the transaction at the end 😉

File: queries.sql [hunk 7]
```diff
@@ -12,7 +12,46 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals SET species = 'unspecified' WHERE species IS NULL;
+
+ROLLBACK;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
+
```
Comment (cynthiainga) [inline queries.sql:32]: - [x] After updating your table, as required in the project you should delete all records in the `animals` table, then roll back the transaction 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Fourteen98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] Good commit messages ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @Fourteen98 , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**